dev/tools/controllerbuilder/pkg/codegen/typegenerator.go

```diff
@@ -120,6 +120,11 @@ func (g *TypeGenerator) WriteVisitedMessages() error {
 			continue
 		}
 
+		if hasOnlyOutputFields(msg) {
+			klog.Infof("skipping generation of %q as it only contains output fields", msg.FullName())
```

Comment:
I see why you want to do this. But it will cause rabbit holes if the existence of the field itself is configurable. PAM entitlement has one case where a field is structured but has no sub fields . I believe this function will skip that field.